### PR TITLE
Fix PCT problem of #8250

### PR DESCRIPTION
**Pull request by daniel-beck**
Amends #8250.

Passed PCT in https://github.com/jenkinsci/bom/pull/2412.

Error looks a lot like what motivated https://github.com/jenkinsci/plugin-pom/pull/693.

### Testing done

Passed PCT in https://github.com/jenkinsci/bom/pull/2412.

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by daniel-beck** [comment]
Looking at the test failures in https://github.com/jenkinsci/bom/pull/2407, almost all failing plugins have servlet-api 3 on their classpath, indicating this is a problem with those plugins, rather than core (also demonstrated by the nature of the change in this PR).

These failing plugins have servlet-api 3 on their classpath:
* apache-httpcomponents-client-4-api
* apache-httpcomponents-client-5-api
* authentication-tokens
* checks-api
* git-server
* handy-uri-templates-2-api
* jakarta-activation-api
* jakarta-mail-api
* javax-activation-api
* javax-mail-api
* jaxb
* jjwt-api
* jsch
* mapdb-api
* pam-auth
* pipeline-github-lib
* pipeline-graph-analysis
* pipeline-milestone-step
* pipeline-stage-step
* plain-credentials
* pubsub-light
* rebuild
* snakeyaml-api
* ssh-agent
* trilead-api
* variant
* workflow-step-api
* ws-cleanup

These failing plugins don't, and they have other test failures:
* [pipeline-groovy-lib](https://github.com/jenkinsci/bom/pull/2407/checks?check_run_id=16152283567) (no clue)
* [script-security](https://github.com/jenkinsci/bom/pull/2407/checks?check_run_id=16152297645) (https://github.com/jenkinsci/jenkins/pull/8412)
* [text-finder](https://github.com/jenkinsci/bom/pull/2407/checks?check_run_id=16152213306) (https://github.com/jenkinsci/jenkins/pull/8412)
* [workflow-durable-task-step](https://github.com/jenkinsci/bom/pull/2407/checks?check_run_id=16152497350) (https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/335)

I have not confirmed that all non-failing plugins lack servlet-api 3 on their classpath, but it's a good guess so far.

So while this workaround fixes PCT runs, longer term whatever's wrong with these plugins that puts servlet-api 3 on their classpath should be fixed.

**Review by timja — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
Thanks!  Approving this because it improves the current situation and seems low risk

**Review by basil — APPROVED** [review]
Good workaround until more plugins are released with a parent POM that includes https://github.com/jenkinsci/plugin-pom/pull/693.

---

This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by timja** [comment]
> I need more education to understand why servlet-api 3 would be on the classpath of the apache-httpcomponents-client-4-api-plugin when running the injected test

as alluded to by Basil, the http client has not had a release in many months and is using an old version of the parent pom:
https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/blob/7ab9d17134a5926b061f8f25a1f7787b2fea990f/pom.xml#L8

**Comment by daniel-beck** [comment]
> should we consider reverting the 404 page improvement until the plugin test fixes are released

Would be completely unjustified.

**Comment by MarkEWaite** [comment]
> > I need more education to understand why servlet-api 3 would be on the classpath of the apache-httpcomponents-client-4-api-plugin when running the injected test
> 
> as alluded to by Basil, the http client has not had a release in many months and is using an old version of the parent pom: https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/blob/7ab9d17134a5926b061f8f25a1f7787b2fea990f/pom.xml#L8

Thanks very much.  That answers my question about resolving the issue in the plugin tests.

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**
